### PR TITLE
adapt pillars of deployment to apply saptune solutions

### PR DESCRIPTION
**Pull request by MalloZup**
## Description:

This PR is a follow-up for the previous work on the saptune module in salt. It enables the mechanism to apply to all node a saptune solution.

For reference look at prs:

- saptune module implementation https://github.com/SUSE/salt-shaptools/pull/45 ( shaptools)
- https://github.com/SUSE/sapnwbootstrap-formula/pull/27 (netweaver formula enablement)
- https://github.com/SUSE/saphanabootstrap-formula/pull/65 (hana formula enablement )

# behaviour changes/features:

This pr , with the new tfvars

`saptune_solution = "HANA"` will allow to apply to all node the saptune solution (in this case) HANA.
So that the user can choose which saptune solution can apply to all (netweaver and ha) cluster nodes .

### CODE REVIEW

**Comment by MalloZup** [comment]
```
bin  hana_inst_media  salt
h-hana01:~ # saptune solution list

All solutions (* denotes enabled solution, O denotes override file exists for solution, D denotes deprecated solutions):
	BOBJ               - 941735 1771258 2578899 SAP_BOBJ
 *	HANA               - 941735 1771258 1980196 2578899 2684254 2382421 2534844
```
works fine

File: doc/saptune.md [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+You can tune your hana or netweaver nodes with saptune during the deployment phase.
+
+Currently we support following features:
+
+- 1) apply a saptune solution during deployement
+
+
+1) Apply a saptune solution during deployement:
+
+In order apply a saptune solution, you need to adapt the pillars:
```
Comment (arbulu89) [inline doc/saptune.md]: 'In order to apply`

File: doc/saptune.md [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+
+```
+saptune_solution: 'HANA'
+```
+
+By default the pillars are configured to apply HANA for hana nodes and NETWEAVER solution for netweavers.
+
+For further information refer to the saphanaboostrap-formula or netweaver.
+The code for module is implemented in https://github.com/SUSE/salt-shaptools
```
Comment (arbulu89) [inline doc/saptune.md]: `The code for the module`

File: salt/netweaver_node/files/pillar/netweaver.sls [hunk 3]
```diff
@@ -11,6 +11,8 @@
 {%- set virtual_host_mask = 24 %}
 {%- endif %}
 
+# apply by default the netweaver solution
+saptune_solution: 'NETWEAVER'
```
Comment (arbulu89) [inline salt/netweaver_node/files/pillar/netweaver.sls]: I think this is incorrect.

`saptune_solution` should come after `netweaver`, right?

Comment (MalloZup) [reply salt/netweaver_node/files/pillar/netweaver.sls]: right https://github.com/SUSE/sapnwbootstrap-formula/blob/master/pillar.example#L43

Comment (MalloZup) [reply salt/netweaver_node/files/pillar/netweaver.sls]: done

**Comment by MalloZup** [comment]
@arbulu89 ok thx for review. changed adressed. :brazil:

File: doc/saptune.md [hunk 4]
```diff
@@ -0,0 +1,21 @@
+# Saptune deployment configuration
+
+You can tune your hana or netweaver nodes with saptune during the deployment phase.
+
+Currently we support following features:
```
Comment (arbulu89) [inline doc/saptune.md]: `we support **the** following`

File: doc/saptune.md [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+
+- 1) apply a saptune solution during deployement
```
Comment (arbulu89) [inline doc/saptune.md]: `s/deployement/deployment/`

File: doc/saptune.md [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+
+
+1) Apply a saptune solution during deployement:
```
Comment (arbulu89) [inline doc/saptune.md]: `s/deployement/deployment/`

File: doc/saptune.md [hunk 7]
```diff
@@ -0,0 +1,21 @@
...
+
+In order to apply a saptune solution, you need to adapt the pillars:
+
+```
+saptune_solution: 'HANA'
+```
+
+By default the pillars are configured to apply HANA for hana nodes and NETWEAVER solution for netweavers.
+
+For further information refer to the saphanaboostrap-formula or netweaver.
```
Comment (arbulu89) [inline doc/saptune.md]: Change `netweaver` by `sapnwbootstrap-formula`

**Pull request merged**